internal/subscription/core.go

```diff
@@ -68,12 +71,38 @@ func (c *Core) CreateSubscription(ctx context.Context, m *Model) error {
 		logger.Ctx(ctx).Errorw("topic project not found", "name", m.ExtractedTopicProjectID)
 		return merror.New(merror.NotFound, "project not found")
 	}
-	if ok, err = c.topicCore.ExistsWithName(ctx, m.Topic); !ok {
+
+	var topicModel *topic.Model
+	if topicModel, err = c.topicCore.Get(ctx, m.Topic); err != nil {
 		if err != nil {
 			return err
 		}
-		return merror.Newf(merror.NotFound, "topic with name %s not found", m.Topic)
 	}
+
+	admin, aerr := c.brokerStore.GetAdmin(ctx, messagebroker.AdminClientOptions{})
+	if aerr != nil {
+		return aerr
+	}
+
+	// create retry topic for subscription
+	_, terr := admin.CreateTopic(ctx, messagebroker.CreateTopicRequest{
+		Name:          m.RetryTopic,
+		NumPartitions: topicModel.NumPartitions,
```

Comment:
@Abhishekvrshny @sunnyagg Should `retry` and `dlq` topics be also created with the same number of partitions as the primary topic?

Reply:
since we listen to retry topic for every steam, we will end up spawning n consumers for retry topic too, i think its better to keep it same. dlq is still debatable, in future we can give control to user for dlq.